src/main/kotlin/org/wycliffeassociates/sourceaudio/common/directory/upload/FilePathGenerator.kt

```diff
@@ -0,0 +1,48 @@
+package org.wycliffeassociates.sourceaudio.common.directory.upload
+
+import org.wycliffeassociates.sourceaudio.common.data.model.*
+import java.lang.IllegalArgumentException
+
+object FilePathGenerator {
+
+    @Throws(IllegalArgumentException::class)
+    fun createPathFromFile(fileModel: FileUploadModel): String {
+        val pathPrefix = getPathPrefix(
+            fileModel.languageCode,
+            fileModel.dublinCoreId,
+            fileModel.projectId,
+            fileModel.extension
+        )
+
+        val isContainer = ContainerExtensions.isSupported(fileModel.extension)
+        val isContainerAndCompressed = isContainer && CompressedExtensions.isSupported(fileModel.mediaExtension)
+        val isFileAndCompressed = !isContainer && CompressedExtensions.isSupported(fileModel.extension)
+
+        return when {
```

Comment:
I like this approach of defining flags and using them to return the correct path structure.